### ISSUE TITLE
GR1ArmsOnly fails to perform accurate grasping in TwoArmCanSortRandom environment with GR00T policy

### ISSUE THREAD

**Issue by 4133223** [NVIDIA/Isaac-GR00T#210]
Description:
 I'm trying to deploy the GR00T-N1.5 policy on the TwoArmCanSortRandom environment using the GR1ArmsOnly robot configuration from dexmimicgen. The robot arm consistently fails to perform successful grasping - it moves to a position above the target object and stops without completing the pick-and-place action (see attached screenshot).

Environment & Configuration:
Simulation Environment: robosuite
Robot: GR1ArmsOnly
Task: TwoArmCanSortRandom
Policy: GR00T-N1.5 (fourier_gr1_arms_only modality)

Observed Behavior:
The right arm reaches above the can but does not lower to grasp it

What I've Tried:
Adjusted camera parameters to match policy's training data perspective
Aligned GR00T's action dimensions with robosuite's requirements

Here is my code：
import os

import torch
import gr00t
from gr00t.model.policy import Gr00tPolicy
from gr00t.experiment.data_config import DATA_CONFIG_MAP
import argparse
import cv2
import numpy as np
import robosuite
from robosuite import load_composite_controller_config
import dexmimicgen

ENV_ROBOTS = {
    "TwoArmThreading": ["Panda", "Panda"],
    "TwoArmThreePieceAssembly": ["Panda", "Panda"],
    "TwoArmTransport": ["Panda", "Panda"],
    "TwoArmLiftTray": ["PandaDexRH", "PandaDexLH"],
    "TwoArmBoxCleanup": ["PandaDexRH", "PandaDexLH"],
    "TwoArmDrawerCleanup": ["PandaDexRH", "PandaDexLH"],
    "TwoArmCoffee": ["GR1FixedLowerBody"],
    "TwoArmPouring": ["GR1FixedLowerBody"],
    "TwoArmCanSortRandom": ["GR1ArmsOnly"],
}
def inverse_format_action(joint_values):
    compressed = np.array([
        np.mean(joint_values[0:2]), 
        np.mean(joint_values[2:4]),  
        np.mean(joint_values[4:6]),  
        np.mean(joint_values[6:8]),  
        np.mean(joint_values[8:10]),  
        joint_values[10]  
    ])

    return compressed

if __name__ == "__main__":
    # Parse arguments
    parser = argparse.ArgumentParser()
    parser.add_argument(
        "--env",
        type=str,
        default="TwoArmCanSortRandom",
        help="Name of the environment to run",
    )
    parser.add_argument(
        "--render",
        default=True,
        help="Whether to render the environment",
    )
    args = parser.parse_args()
    assert args.env in ENV_ROBOTS, f"Environment {args.env} not found!"
    env_kwargs = {
        "env_name": args.env,
        "robots": ENV_ROBOTS[args.env],
        "controller_configs": load_composite_controller_config(
            robot=ENV_ROBOTS[args.env][0]
        ),
        "has_renderer": '--render',
        "has_offscreen_renderer": True,
        "ignore_done": True,
        "use_camera_obs": True,
        "control_freq": 20,
    }
    env = robosuite.make(
        **env_kwargs,
    )
    env.reset()
    if args.render:
        env.render()
    action_vector = np.zeros(26)
    # action_vector = np.random.randn(*env.action_spec[0].shape) * 0.1
    ###################################################################################################################
    MODEL_PATH = "/home/user1/GR00T-N1.5-3B"
    # REPO_PATH is the path of the pip install gr00t repo and one level up
    REPO_PATH = os.path.dirname(os.path.dirname(gr00t.__file__))
    DATASET_PATH = os.path.join(REPO_PATH, "demo_data/robot_sim.PickNPlace")
    EMBODIMENT_TAG = "gr1"
    DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
    data_config = DATA_CONFIG_MAP['fourier_gr1_arms_only']
    modality_config = data_config.modality_config()
    modality_transform = data_config.transform()
    policy = Gr00tPolicy(
        model_path=MODEL_PATH,
        embodiment_tag=EMBODIMENT_TAG,
        modality_config=modality_config,
        modality_transform=modality_transform,
        device=DEVICE,
    )
  ####################################################################################################################
    for i in range(1000):
        obs, reward, done, _ = env.step(action_vector)
        #image
        eye_image = obs['agentview_image']
        eye_image = cv2.flip(eye_image, 0)
        eye_image = cv2.cvtColor(eye_image, cv2.COLOR_RGB2BGR)
        eye_image  = cv2.resize(eye_image, (256, 256))
        cv2.imshow("eye_image", eye_image)
        cv2.waitKey(1)

        #joint_state
        right_arm_joint_state = obs['robot0_joint_pos'][0:7]
        left_arm_joint_state = obs['robot0_joint_pos'][7:14]
        right_gripper_state = inverse_format_action(obs['robot0_right_gripper_qpos'])
        left_gripper_state = inverse_format_action(obs['robot0_left_gripper_qpos'])

        #desscription
        des = 'pick the red can and put it in the red box '
        if i == 0 :
            # action
            right_arm_joint_action = action_vector[0:7]
            left_arm_joint_action = action_vector[7:14]
            right_gripper_action = action_vector[14:20]
            left_gripper_action = action_vector[20:26]

            step_data = {'video.ego_view': eye_image.reshape(1, 256, 256, 3),
                         'state.left_arm': left_arm_joint_state.reshape(1,7),
                         'state.right_arm': right_arm_joint_state.reshape(1,7),
                        'state.left_hand': left_gripper_state.reshape(1,6),
                         'state.right_hand':  right_gripper_state.reshape(1,6),
                         'action.left_arm':  np.tile(left_arm_joint_action, (1, 1)) ,
                          'action.right_arm': np.tile(right_arm_joint_action, (1, 1)) ,
                         'action.left_hand': np.tile(left_gripper_action ,(1, 1)),
                         'action.right_hand': np.tile(right_gripper_action, (1, 1)),
                        'annotation.human.action.task_description':[des]
                   }
        else:
            step_data = {'video.ego_view': eye_image.reshape(1, 256, 256, 3),
                         'state.left_arm': left_arm_joint_state.reshape(1, 7),
                         'state.right_arm': right_arm_joint_state.reshape(1, 7),
                         'state.left_hand': left_gripper_state.reshape(1, 6),
                         'state.right_hand': right_gripper_state.reshape(1, 6),
                         'action.left_arm':  predicted_action['action.left_arm'],
                         'action.right_arm': predicted_action['action.right_arm'],
                         'action.left_hand': predicted_action['action.left_hand'],
                         'action.right_hand': predicted_action['action.right_hand'] ,
                         'annotation.human.action.task_description': [des]
                         }

        predicted_action = policy.get_action(step_data)
        action_vector = np.concatenate([
           predicted_action['action.right_arm'][0],
            predicted_action['action.left_arm'][0],
            predicted_action['action.right_hand'][0],
            predicted_action['action.left_hand'][0],
        ], axis=0)

        if args.render:
            env.render()
 Attached Screenshot:

![Image](https://github.com/user-attachments/assets/c0777e1f-2dee-44f7-9158-9f6ead305cc2)

**Comment by HuFY-dev** [comment]
Please check out the sim we recently released for evaluation: https://github.com/robocasa/robocasa-gr1-tabletop-tasks

**Comment by lishisen01** [comment]
I have the same problem. Have you solved it?

**Comment by chengyuehuang511** [comment]
I think it's probably because you need to finetune the base model on dexmg? However they did not provide the fine-tuned ckpt
